src/words.jl

```diff
@@ -33,31 +33,37 @@ If type is not specified in the constructor it will default to `Int16`.
 """
 struct Word{T} <: AbstractWord{T}
     ptrs::Vector{T}
+
+    function Word{T}(v::AbstractVector{<:Integer}) where {T}
+        @assert all(x -> x > 0, v) "All entries of a Word must be positive integers"
+        return new{T}(v)
+    end
+    Word{T}(v::AbstractVector{<:Unsigned}) where {T<:Unsigned} = new{T}(v)
 end
 
 # setting the default type to Int16
-Word(x::Union{<:Vector{<:Integer}, <:AbstractVector{<:Integer}}) = Word{Int16}(x)
+Word(x::Union{<:Vector{<:Integer},<:AbstractVector{<:Integer}}) = Word{UInt16}(x)
```

Comment:
Well, I think that something is wrong with the requirement that all the elements of a word must be unsigned integers, but I cannot tell you exactly what it is. I need to think about that.

By the way, am I seeing it correctly that for a vector of integers it is not allowed for any pointer to be less or equal to 0, but in case of unsigned integers it is allowed for a pointer to be 0?

Reply:
yes, that was a premature optimization